File: 7 kyu/(7 kyu) Growth of a Population/(7 kyu) Growth of a Population.go
package kata

import "math"

func NbYear(population int, percent float64, aug int, populationToSurpass int) int {
  var years int = 0

  for ok := true; ok; ok = population < populationToSurpass {
    population += int(math.Floor(float64(population) * (percent / float64(100)) + float64(aug)))
    years += 1
  }

  return years
}

File: 7 kyu/(7 kyu) Growth of a Population/(7 kyu) Growth of a Population.dart
int nbYear(int population, double percent, int aug, int populationToSurpass) {
  var years = 0;

  do {
    population += (population * (percent / 100) + aug).floor();
    years += 1;
  } while (population < populationToSurpass);

  return years;
}
